File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/exam/CreateExamRequest.java
package org.example.olympic_ot_project.dto.exam;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class CreateExamRequest {
    @NotBlank(message = "Tên cuộc thi không được để trống")
    private String name;
    @NotNull(message = "Người ra đề không được để trống")
    private Integer createdById;
    private boolean shuffleOption;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/exam/create_exam_request.dart
class CreateExamRequest {
  final String name;
  final int createdById;
  final bool shuffleOption;

  CreateExamRequest({
    required this.name,
    required this.createdById,
    required this.shuffleOption,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'createdById': createdById,
      'shuffleOption': shuffleOption,
    };
  }
}
